app/models/student_profile.py

```diff
@@ -1,36 +1,81 @@
-    db.Column('student_profile_id', db.Integer, db.ForeignKey('student_profile.id'))
-)
+                          db.Column('major_id', db.Integer,
+                                    db.ForeignKey('major.id')),
+                          db.Column('student_profile_id', db.Integer,
+                                    db.ForeignKey('student_profile.id'))
+                          )
 
 
 class StudentProfile(db.Model):
     id = db.Column(db.Integer, primary_key=True)
-    user_id = db.Column(db.Integer, db.ForeignKey('users.id'))
     user = db.relationship("User", back_populates="student_profile")
     # PERSONAL INFO
     high_school = db.Column(db.String, index=True)
     district = db.Column(db.String, index=True)
     city = db.Column(db.String, index=True)
-    state = db.Column(db.String(2), index=True)
-    graduation_year = db.Column(db.Integer, index=True)
-    grade = db.Column(db.Integer, index=True)  
+    state = db.Column(db.String, index=True)
+    graduation_year = db.Column(db.String, index=True)
+    grade = db.Column(db.Integer, index=True)
     # ACADEMIC INFO
     gpa = db.Column(db.Float, index=True)
-    test_scores = db.relationship('TestScore', backref='student_profile', lazy=True)
+    test_scores = db.relationship(
+        'TestScore', backref='student_profile', lazy=True)
     majors = db.relationship('Major', secondary=student_majors,
-        backref=db.backref('student_profiles', lazy='dynamic'))
-    colleges = db.relationship('College', secondary=student_colleges, 
-        backref=db.backref('student_profiles', lazy='dynamic'))
+                             backref=db.backref('student_profiles', lazy='dynamic'))
+    colleges = db.relationship('College', secondary=student_colleges,
+                               backref=db.backref('student_profiles', lazy='dynamic'))
     # APPLICATION INFO
     common_app_essay = db.Column(db.String, index=True)
-    essays = db.relationship('Essay', backref='student_profile', lazy=True)
-    recommendation_letters = db.relationship('RecommendationLetter', backref='student_profile', lazy=True)
\ No newline at end of file
+    essays = db.relationship('Essay')
+    recommendation_letters = db.relationship('RecommendationLetter')
+
+    @staticmethod
+    def generate_fake():
+        fake = Faker()
+        year = random.choice([['2018', '12'], ['2019', '11'], ['2020', '10']])
+        profile = StudentProfile(
+            high_school='{} High School'.format(fake.street_name()),
+            district='{} District'.format(fake.city()),
+            city=fake.city(),
+            state=fake.state(),
+            graduation_year=year[0],
+            grade=year[1],
+            gpa=round(random.uniform(2, 4), 2),
+            test_scores=TestScore.generate_fake(),
+            majors=random.sample(Major.query.all(), 3),
+            colleges=random.sample(College.query.all(), 3),
+            common_app_essay='Common App Essay Link',
```

Comment:
As a future note, if we do stick with links, it might be good to change this variable name (common_app_essay) to make this clearer in the code?

Reply:
Yes, for sure, just didn't know what to link it to. I'll make a little todo for this!

Reply:
Got it, that makes sense.

Reply:
For now I'll just add a comment.